kobo/hub/views.py

```diff
@@ -196,12 +174,35 @@ def task_log(request, id, log_name):
         "content": content,
         "log_name": log_name,
         "task": task,
-        "json_url": reverse("task/log-json", args=[id, log_name]),
+        "json_url": reverse("task/log-json", args=[task.id, log_name]),
     }
 
     return render_to_response("task/log.html", context, context_instance=RequestContext(request))
 
 
+def task_log(request, id, log_name):
+    """
+    IMPORTANT: reverse to 'task/log-json' *must* exist
+    """
+    if os.path.basename(log_name).startswith("traceback") and not request.user.has_perm('hub.can_see_traceback'):
+        return HttpResponseForbidden("You don't have permission to see the traceback.")
+
+    task = get_object_or_404(Task, id=id)
+
+    file_path = task.logs._get_absolute_log_path(log_name)
+    if not os.path.isfile(file_path) and not file_path.endswith(".gz"):
+        file_path = task.logs._get_absolute_log_path(log_name + ".gz")
```

Comment:
Wouldn't this also fall into that failure case where log_name + .gz does not exist as well? So it might be worth checking for that failure case instead of having the failure buried further down in "log doesn't exist"

Reply:
I haven't changed this part of the code and would prefer to avoid touching it it without a specific reason to do so.

I did a quick check and I can confirm that, in the raw mode, the error handling is broken both on kobo 0.5.0 and as of this commit, and have filed it as https://github.com/release-engineering/kobo/issues/49 - it actually doesn't fail if the log file is missing.  I don't want to try fixing that since that could break clients (e.g. a client who creates a task and then immediately starts polling for the log file before it might even exist).

Reply:
Actually as weird as it sounds, maybe that's even intentional?  That might explain the content_len = 0 code path when a file doesn't exist, which I'd wondered about...

Reply:
Yeah could be, just seems an odd way to sprinkle all this logic around that attempts to operate on a file but nothing existing.